### ISSUE TITLE
Revise/Improve the CMake build

### ISSUE THREAD

**Issue by carstene1ns** [EasyRPG/liblcf#215]
Currently CMake seems not really supported, as it stays in a subfolder.
We should:
- [ ] move `CMakeLists.txt` to the root folder
- [ ] improve the file by removing deprecated/unneeded stuff
- [ ] upgrade Syntax¹

1) We currently use 2.8, however 3.0 has been released 2014, so things could be [tweaked](https://cmake.org/cmake/help/v3.0/release/3.0.0.html#new-features) (e.g. [CMP0048](https://cmake.org/cmake/help/v3.0/policy/CMP0048.html#policy:CMP0048)).

**Comment by Ghabry** [comment]
One 3.x version added FindICU, but couldn't figure out yet which version. This would be the minimal version imo.

**Comment by carstene1ns** [comment]
Changelog says 3.7, which is pretty recent (Nov 2016). I think we should just ship it ourselves.

**Comment by Ghabry** [comment]
Yeah thats, too new. Good option assuming the FindICU script is compatible with 3.0 :D

**Comment by carstene1ns** [comment]
Debian Jessie (oldstable) had 3.0.2, so this should be the absolute minimum for us.
I vote for 3.6, as this is available as backport there and stretch has already 3.7.

**Comment by Ghabry** [comment]
okay than 3.6 is fine for me when even Debian is up-to-date here :D.
Because 3.7 is already so much used I vote against bundling the new FindICU script and to add a conditional instead that checks for < 3.7 and uses pkg-config (as we already do) in that case and for >= 3.7 the new fancy script.

**Comment by carstene1ns** [comment]
Merged the PR, updated the checkmarks.
However, we have now a broken `liblcf-config.cmake`... :see_no_evil:
Build log excerpt:
```
-- Found liblcf: /usr/lib/liblcf.so (found version "0.5.3") 
CMake Error at /usr/lib64/cmake/liblcf/liblcf-config.cmake:88 (target_compile_features):
  target_compile_features called with non-compilable target type
Call Stack (most recent call first):
  CMakeLists.txt:220 (find_package)
```

This breaks here: https://github.com/EasyRPG/liblcf/blob/master/builds/liblcf-config.cmake.in#L88

From https://cmake.org/cmake/help/v3.7/command/target_compile_features.html (emphasis mine):
>The named <target> must have been created by a command such as add_executable() or add_library() and **must not be an IMPORTED target**.
